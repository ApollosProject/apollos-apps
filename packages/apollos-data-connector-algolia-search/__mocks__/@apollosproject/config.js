const ApolloServer = require.requireActual('@apollosproject/config').default;

ApolloServer.loadJs({
  ROCK: {
    TIMEZONE: 'America/New_York',
  },
  ALGOLIA: {
    APPLICATION_ID: 'some-application-id',
    API_KEY: 'some-api-key',
    CONFIGURATION: {},
  },
});

export default ApolloServer;
