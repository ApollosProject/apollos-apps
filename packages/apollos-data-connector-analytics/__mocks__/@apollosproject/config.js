const ApollosConfig = require.requireActual('@apollosproject/config');

const defaults = {
  ANALYTICS: { SEGMENT_KEY: 'something', GA_ID: 'something-else' },
};

ApollosConfig.default.loadJs(defaults);

console.log(ApollosConfig);

class dataSource {
  constructor() {
    this.config = null;
    this.loadJs = this.loadJs.bind(this);
  }

  initialize({ context }) {
    if (context?.church?.slug) {
      const config = new ApollosConfig.Config();
      config.loadJs(defaults);
      this.config = config;
      this.attachConfigToClass(this.config);
    }
  }

  loadJs(input) {
    this.config.loadJs(input);
    this.attachConfigToClass(this.config);
  }

  attachConfigToClass(config) {
    Object.keys({ ...config }).forEach((key) => {
      if (key !== 'config') {
        Object.defineProperty(this, key, {
          enumerable: true,
          configurable: true,
          writable: true,
          value: config[key],
        });
      }
    });
  }
}

export { dataSource };
export default ApollosConfig;
