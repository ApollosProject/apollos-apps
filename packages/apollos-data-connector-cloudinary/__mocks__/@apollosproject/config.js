const ApollosConfig = require.requireActual('@apollosproject/config');

const defaults = {
  CLOUDINARY: {
    URL: 'cloudinary://123456789012345:abcdeghijklmnopqrstuvwxyz12@n07t21i7',
  },
};

ApollosConfig.default.loadJs(defaults);

class dataSource {
  constructor() {
    this.config = null;
    this.loadJs = this.loadJs.bind(this);
  }

  initialize({ context }) {
    if (context?.church?.slug) {
      const config = new ApollosConfig.Config();
      config.loadJs(defaults);
      this.config = config;
      this.attachConfigToClass(this.config);
    }
  }

  loadJs(input) {
    this.config.loadJs(input);
    this.attachConfigToClass(this.config);
  }

  attachConfigToClass(config) {
    Object.keys({ ...config }).forEach((key) => {
      if (key !== 'config') {
        Object.defineProperty(this, key, {
          enumerable: true,
          configurable: true,
          writable: true,
          value: config[key],
        });
      }
    });
  }
}

export { dataSource };
export default ApollosConfig;
