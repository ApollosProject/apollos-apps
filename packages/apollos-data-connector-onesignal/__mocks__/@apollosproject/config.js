const ApolloServer = require.requireActual('@apollosproject/config').default;

ApolloServer.loadJs({
  ONE_SIGNAL: {
    APP_ID: 'some-app-id',
  },
});

export default ApolloServer;
