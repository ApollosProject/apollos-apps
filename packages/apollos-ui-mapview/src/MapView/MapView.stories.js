/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import { storiesOf } from '@apollosproject/ui-storybook';

import MapView from '.';

const campuses = [
  {
    city: 'Richardson',
    distanceFromLocation: null,
    id: 'Campus:965b6e6d7046a885bea4e300b5c0400d',
    image: 'https://www.placecage.com/c/220/220',
    latitude: 32.95103,
    longitude: -96.74738,
    name: 'Dallas Campus',
    postalCode: '75080-5525',
    state: 'TX',
    street1: '102 N Weatherred Dr',
  },
  {
    city: 'Cincinnati',
    distanceFromLocation: 2037.6461577685534,
    id: 'Campus:4f68015ba18662a7409d1219a4ce013e',
    image: 'https://www.placecage.com/c/220/220',
    latitude: 39.10501,
    longitude: -84.51138,
    name: 'Cincinnati Campus',
    postalCode: '45202-2118',
    state: 'OH',
    street1: '120 E 8th St',
  },
];

const currentCampus = {
  city: 'Richardson',
  distanceFromLocation: null,
  id: 'Campus:965b6e6d7046a885bea4e300b5c0400d',
  image: 'https://www.placecage.com/c/220/220',
  latitude: 32.95103,
  longitude: -96.74738,
  name: 'Dallas Campus',
  postalCode: '75080-5525',
  state: 'TX',
  street1: '102 N Weatherred Dr',
};

const initialRegion = {
  latitude: 39.809734,
  latitudeDelta: 100,
  longitude: -98.555618,
  longitudeDelta: 46.205357142857146,
};

const navigation = {
  getParam: () => {},
  navigate: () => {},
  goBack: () => {},
};

storiesOf('ui-mapview/MapView', module)
  .add('default', () => (
    <MapView
      navigation={navigation}
      campuses={campuses}
      initialRegion={initialRegion}
      userLocation={currentCampus}
      currentCampus={currentCampus}
      onLocationSelect={async () => {}}
    />
  ))
  .add('buttonTitle', () => (
    <MapView
      navigation={navigation}
      campuses={campuses}
      initialRegion={initialRegion}
      userLocation={currentCampus}
      currentCampus={currentCampus}
      onLocationSelect={async () => {}}
      buttonTitle={'Custom buttonTitle'}
    />
  ))
  .add('isLoading', () => (
    <MapView
      navigation={navigation}
      campuses={campuses}
      initialRegion={initialRegion}
      userLocation={currentCampus}
      currentCampus={currentCampus}
      onLocationSelect={async () => {}}
      isLoading
    />
  ));
