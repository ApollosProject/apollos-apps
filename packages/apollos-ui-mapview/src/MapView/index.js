import React, { Component } from 'react';
import PropTypes from 'prop-types';
import RNMapView from 'react-native-maps';
import { Animated, Dimensions, Platform, PixelRatio } from 'react-native';
import { SafeAreaView } from 'react-navigation';
import { debounce } from 'lodash';

import {
  Button,
  Touchable,
  PaddedView,
  FlexedView,
  styled,
  withTheme,
  CampusCard,
  withIsLoading,
} from '@apollosproject/ui-kit';
import { MediaPlayerSpacer } from '@apollosproject/ui-media-player';

import Marker from '../Marker';

export const FlexedMapView = styled({ flex: 1 })(({ mapRef, ...props }) => (
  <RNMapView ref={mapRef} {...props} />
));

const Footer = styled(
  {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
  },
  'ui-mapview.MapView.Footer'
)(SafeAreaView);

const StyledCampusCard = styled(
  ({ theme, cardWidth }) => ({
    width: cardWidth,
    marginHorizontal: theme.sizing.baseUnit / 4,
  }),
  'ui-mapview.MapView.StyledCampusCard'
)(CampusCard);

class MapView extends Component {
  static propTypes = {
    campuses: PropTypes.arrayOf(
      PropTypes.shape({
        latitude: PropTypes.number.isRequired,
        longitude: PropTypes.number.isRequired,
      })
    ),
    currentCampus: PropTypes.shape({
      latitude: PropTypes.number.isRequired,
      longitude: PropTypes.number.isRequired,
    }),
    onLocationSelect: PropTypes.func.isRequired,
    initialRegion: PropTypes.shape({
      latitude: PropTypes.number.isRequired,
      longitude: PropTypes.number.isRequired,
      latitudeDelta: PropTypes.number,
      longitudeDelta: PropTypes.number,
    }).isRequired,
    userLocation: PropTypes.shape({
      latitude: PropTypes.number.isRequired,
      longitude: PropTypes.number.isRequired,
    }),
    theme: PropTypes.shape({
      sizing: PropTypes.shape({
        baseUnit: PropTypes.number,
      }),
    }),
    navigation: PropTypes.shape({
      goBack: PropTypes.func,
    }),
    // eslint-disable-next-line react/no-unused-prop-types
    Marker: PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.func,
      PropTypes.object, // type check for React fragments
    ]),
    isLoading: PropTypes.bool,
    isLoadingSelectedCampus: PropTypes.bool,
  };

  static defaultProps = {
    Marker,
  };

  constructor(props) {
    super();
    // sets width of cards to be full width of the screen minus a bit for spacing
    this.cardWidth =
      Dimensions.get('window').width - props.theme.sizing.baseUnit * 2.25;

    this.animation = new Animated.Value(0);
    this.scrollView = null;

    // We need to cache this value in state, because otherwise the campus order changes
    // After selecting a campus, but before the home feed loads.
    this.state = { currentCampus: null };

    this.cardScrollPositionOffset =
      this.cardWidth + props.theme.sizing.baseUnit * 0.5;
  }

  componentDidMount() {
    this.animation.addListener(debounce(this.updateCoordinates));
  }

  componentDidUpdate(oldProps) {
    // update mapview if there are campuses and the location changes
    if (this.props.currentCampus && !this.state.currentCampus) {
      // eslint-disable-next-line react/no-did-update-set-state
      this.setState({ currentCampus: this.props.currentCampus });
    }
    if (
      this.props.campuses.length &&
      oldProps.userLocation !== this.props.userLocation
    ) {
      this.updateCoordinates({ value: this.previousScrollPosition });
    }
  }

  componentWillUnmount() {
    this.animation.removeListener(this.updateCoordinates);
  }

  get currentCampus() {
    const cardIndex = Math.floor(
      this.previousScrollPosition / this.cardWidth + 0.3
    ); // animate 30% away from landing on the next item;
    return this.sortedCampuses[cardIndex];
  }

  get sortedCampuses() {
    const { campuses = [] } = this.props;
    const { currentCampus } = this.state;
    if (!currentCampus) {
      return campuses;
    }
    return [
      campuses.find(({ id }) => id === currentCampus.id),
      ...campuses.filter(({ id }) => id !== currentCampus.id),
    ];
  }

  getCampusAddress = (campus) =>
    `${campus.street1}\n${campus.city}, ${campus.state} ${campus.postalCode}`;

  scrollToIndex = (index) => {
    const cardScrollPosition = index * this.cardScrollPositionOffset;

    this.scrollView.getNode().scrollTo({
      x: cardScrollPosition,
      y: 0,
      animated: true,
    });
    this.updateCoordinates({ value: cardScrollPosition });
  };

  updateCoordinates = ({ value }) => {
    this.previousScrollPosition = value;

    const { userLocation } = this.props;
    // campus card height + some padding

    const bottomPadding = this.props.theme.sizing.baseUnit * 18.25;
    const edgePadding = {
      top: this.props.theme.sizing.baseUnit * 6.25,
      left: this.props.theme.sizing.baseUnit * 6.25,
      right: this.props.theme.sizing.baseUnit * 6.25,
      bottom:
        Platform.OS === 'android'
          ? // NOTE: android bug
            // https://github.com/react-native-community/react-native-maps/issues/2543
            PixelRatio.getPixelSizeForLayoutSize(bottomPadding)
          : bottomPadding,
    };

    const visibleCampuses = [
      userLocation,
      ...(this.currentCampus ? [this.currentCampus] : this.sortedCampuses),
    ];

    this.map.fitToCoordinates(visibleCampuses, {
      edgePadding,
    });
  };

  render() {
    const { onLocationSelect, isLoading, isLoadingSelectedCampus } = this.props;
    const interpolations = this.sortedCampuses.map((marker, index) => {
      const inputRange = [
        (index - 1) * this.cardWidth,
        index * this.cardWidth,
        (index + 1) * this.cardWidth,
      ];
      const opacity = this.animation.interpolate({
        inputRange,
        outputRange: [0.35, 1, 0.35],
        extrapolate: 'clamp',
      });
      return { opacity };
    });

    return (
      <FlexedView>
        <FlexedMapView
          initialRegion={this.props.initialRegion}
          showsUserLocation
          mapRef={(map) => {
            this.map = map;
          }}
        >
          {this.sortedCampuses.map((campus, index) => {
            const campusOpacity = {
              opacity: interpolations[index].opacity,
            };
            return (
              <this.props.Marker
                onPress={() => this.scrollToIndex(index)}
                key={campus.id}
                opacityStyle={campusOpacity}
                latitude={campus.latitude}
                longitude={campus.longitude}
              />
            );
          })}
        </FlexedMapView>
        <Footer>
          <Animated.ScrollView
            horizontal
            showsHorizontalScrollIndicator={false}
            snapToInterval={this.cardScrollPositionOffset}
            snapToAlignment={'start'}
            decelerationRate={'fast'}
            contentContainerStyle={{
              paddingHorizontal: this.props.theme.sizing.baseUnit * 0.75,
            }}
            ref={(ref) => (this.scrollView = ref)} // eslint-disable-line
            scrollEventThrottle={16} // roughtly 1000ms/60fps = 16ms
            onScroll={Animated.event(
              [
                {
                  nativeEvent: {
                    contentOffset: {
                      x: this.animation,
                    },
                  },
                },
              ],
              { useNativeDriver: true }
            )}
          >
            {this.sortedCampuses.map((campus) => (
              <Touchable
                key={campus.id}
                onPress={() => onLocationSelect(campus)}
              >
                <StyledCampusCard
                  distance={campus.distanceFromLocation}
                  title={campus.name}
                  description={this.getCampusAddress(campus)}
                  images={[campus.image]}
                  cardWidth={this.cardWidth}
                  isLoading={isLoading}
                />
              </Touchable>
            ))}
          </Animated.ScrollView>
          <MediaPlayerSpacer>
            <PaddedView>
              <Button
                title="Select Campus"
                pill={false}
                type="secondary"
                onPress={() =>
                  onLocationSelect(this.currentCampus || this.sortedCampuses[0])
                }
                loading={isLoadingSelectedCampus}
              />
            </PaddedView>
          </MediaPlayerSpacer>
        </Footer>
      </FlexedView>
    );
  }
}

export default withTheme()(withIsLoading(MapView));
