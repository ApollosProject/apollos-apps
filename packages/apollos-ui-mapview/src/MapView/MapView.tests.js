import React from 'react';
import renderer from 'react-test-renderer';

import { Providers } from '@apollosproject/ui-kit';

import MapView from '.';

const campuses = [
  {
    city: 'Richardson',
    distanceFromLocation: null,
    id: 'Campus:965b6e6d7046a885bea4e300b5c0400d',
    image: 'https://www.placecage.com/c/250/250',
    latitude: 32.95103,
    longitude: -96.74738,
    name: 'Dallas Campus',
    postalCode: '75080-5525',
    state: 'TX',
    street1: '102 N Weatherred Dr',
  },
  {
    city: 'Cincinnati',
    distanceFromLocation: 2037.6461577685534,
    id: 'Campus:4f68015ba18662a7409d1219a4ce013e',
    image: 'https://www.placecage.com/c/250/250',
    latitude: 39.10501,
    longitude: -84.51138,
    name: 'Cincinnati Campus',
    postalCode: '45202-2118',
    state: 'OH',
    street1: '120 E 8th St',
  },
];

const currentCampus = {
  city: 'Richardson',
  distanceFromLocation: null,
  id: 'Campus:965b6e6d7046a885bea4e300b5c0400d',
  image: 'https://www.placecage.com/c/250/250',
  latitude: 32.95103,
  longitude: -96.74738,
  name: 'Dallas Campus',
  postalCode: '75080-5525',
  state: 'TX',
  street1: '102 N Weatherred Dr',
};

const initialRegion = {
  latitude: 39.809734,
  latitudeDelta: 100,
  longitude: -98.555618,
  longitudeDelta: 46.205357142857146,
};

const navigation = {
  getParam: () => {},
  navigate: () => {},
  goBack: () => {},
};

describe('<MapView>', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <MapView
          navigation={navigation}
          campuses={campuses}
          initialRegion={initialRegion}
          userLocation={currentCampus}
          currentCampus={currentCampus}
          onLocationSelect={jest.fn()}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with a custom buttonTitle', () => {
    const tree = renderer.create(
      <Providers>
        <MapView
          navigation={navigation}
          campuses={campuses}
          initialRegion={initialRegion}
          userLocation={currentCampus}
          currentCampus={currentCampus}
          onLocationSelect={jest.fn()}
          buttonTitle={'Custtom buttonTitle'}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
