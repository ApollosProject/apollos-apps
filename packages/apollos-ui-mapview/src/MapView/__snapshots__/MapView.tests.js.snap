// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapView> should render 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 20,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "paddingHorizontal": 12,
        }
      }
      decelerationRate="fast"
      horizontal={true}
      onScroll={[Function]}
      scrollEventThrottle={16}
      showsHorizontalScrollIndicator={false}
      snapToAlignment="start"
      snapToInterval={722}
    >
      <View>
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            cardWidth={714}
            inHorizontalList={true}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 16,
                "margin": 8,
                "marginHorizontal": 4,
                "shadowColor": "rgba(165, 165, 165, 0.6)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 6,
                "width": 714,
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 16,
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexWrap": "nowrap",
                    "height": 96,
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  animate={null}
                  customAnimate={null}
                  onReady={false}
                  style={
                    Object {
                      "aspectRatio": 1,
                      "backgroundColor": "#A5A5A5",
                      "height": "100%",
                      "resizeMode": "cover",
                      "width": "100%",
                    }
                  }
                >
                  <Image
                    onLoad={[Function]}
                    source={
                      Array [
                        Object {
                          "cache": "force-cache",
                          "uri": "https://www.placecage.com/c/250/250",
                          "url": "https://www.placecage.com/c/250/250",
                        },
                      ]
                    }
                    style={
                      Object {
                        "aspectRatio": 1,
                        "backgroundColor": "#A5A5A5",
                        "height": "100%",
                        "opacity": 0,
                        "resizeMode": "cover",
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "paddingHorizontal": 16,
                      "paddingVertical": 16,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "stretch",
                        "flexDirection": "row",
                        "flexWrap": "nowrap",
                        "height": 21,
                        "justifyContent": "space-between",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#303030",
                          "fontFamily": "InterUI-Medium",
                          "fontSize": 14,
                          "lineHeight": 21,
                        }
                      }
                    >
                      Dallas Campus
                    </Text>
                  </View>
                  <Text
                    style={
                      Object {
                        "color": "#505050",
                        "fontFamily": "InterUI-Bold",
                        "fontSize": 12,
                        "lineHeight": 18,
                      }
                    }
                  >
                    102 N Weatherred Dr
Richardson, TX 75080-5525
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            cardWidth={714}
            inHorizontalList={true}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 16,
                "margin": 8,
                "marginHorizontal": 4,
                "shadowColor": "rgba(165, 165, 165, 0.6)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 6,
                "width": 714,
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 16,
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexWrap": "nowrap",
                    "height": 96,
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  animate={null}
                  customAnimate={null}
                  onReady={false}
                  style={
                    Object {
                      "aspectRatio": 1,
                      "backgroundColor": "#A5A5A5",
                      "height": "100%",
                      "resizeMode": "cover",
                      "width": "100%",
                    }
                  }
                >
                  <Image
                    onLoad={[Function]}
                    source={
                      Array [
                        Object {
                          "cache": "force-cache",
                          "uri": "https://www.placecage.com/c/250/250",
                          "url": "https://www.placecage.com/c/250/250",
                        },
                      ]
                    }
                    style={
                      Object {
                        "aspectRatio": 1,
                        "backgroundColor": "#A5A5A5",
                        "height": "100%",
                        "opacity": 0,
                        "resizeMode": "cover",
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "paddingHorizontal": 16,
                      "paddingVertical": 16,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "stretch",
                        "flexDirection": "row",
                        "flexWrap": "nowrap",
                        "height": 21,
                        "justifyContent": "space-between",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#303030",
                          "fontFamily": "InterUI-Medium",
                          "fontSize": 14,
                          "lineHeight": 21,
                        }
                      }
                    >
                      Cincinnati Campus
                    </Text>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                          "justifyContent": "flex-end",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#505050",
                            "fontFamily": "InterUI-Bold",
                            "fontSize": 12,
                            "lineHeight": 18,
                          }
                        }
                      >
                        2038
                        mi
                      </Text>
                    </View>
                  </View>
                  <Text
                    style={
                      Object {
                        "color": "#505050",
                        "fontFamily": "InterUI-Bold",
                        "fontSize": 12,
                        "lineHeight": 18,
                      }
                    }
                  >
                    120 E 8th St
Cincinnati, OH 45202-2118
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "paddingHorizontal": 16,
          "paddingVertical": 16,
        }
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          bordered={false}
          disabled={false}
          pill={false}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#17B582",
              "borderColor": "#17B582",
              "borderRadius": 16,
              "borderWidth": 2,
              "elevation": 2,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "center",
              "opacity": 1,
              "overflow": "hidden",
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "InterUI-Bold",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
          >
            Confirm Selection
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<MapView> should render with a custom buttonTitle 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 20,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "paddingHorizontal": 12,
        }
      }
      decelerationRate="fast"
      horizontal={true}
      onScroll={[Function]}
      scrollEventThrottle={16}
      showsHorizontalScrollIndicator={false}
      snapToAlignment="start"
      snapToInterval={722}
    >
      <View>
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            cardWidth={714}
            inHorizontalList={true}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 16,
                "margin": 8,
                "marginHorizontal": 4,
                "shadowColor": "rgba(165, 165, 165, 0.6)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 6,
                "width": 714,
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 16,
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexWrap": "nowrap",
                    "height": 96,
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  animate={null}
                  customAnimate={null}
                  onReady={false}
                  style={
                    Object {
                      "aspectRatio": 1,
                      "backgroundColor": "#A5A5A5",
                      "height": "100%",
                      "resizeMode": "cover",
                      "width": "100%",
                    }
                  }
                >
                  <Image
                    onLoad={[Function]}
                    source={
                      Array [
                        Object {
                          "cache": "force-cache",
                          "uri": "https://www.placecage.com/c/250/250",
                          "url": "https://www.placecage.com/c/250/250",
                        },
                      ]
                    }
                    style={
                      Object {
                        "aspectRatio": 1,
                        "backgroundColor": "#A5A5A5",
                        "height": "100%",
                        "opacity": 0,
                        "resizeMode": "cover",
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "paddingHorizontal": 16,
                      "paddingVertical": 16,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "stretch",
                        "flexDirection": "row",
                        "flexWrap": "nowrap",
                        "height": 21,
                        "justifyContent": "space-between",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#303030",
                          "fontFamily": "InterUI-Medium",
                          "fontSize": 14,
                          "lineHeight": 21,
                        }
                      }
                    >
                      Dallas Campus
                    </Text>
                  </View>
                  <Text
                    style={
                      Object {
                        "color": "#505050",
                        "fontFamily": "InterUI-Bold",
                        "fontSize": 12,
                        "lineHeight": 18,
                      }
                    }
                  >
                    102 N Weatherred Dr
Richardson, TX 75080-5525
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            cardWidth={714}
            inHorizontalList={true}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 16,
                "margin": 8,
                "marginHorizontal": 4,
                "shadowColor": "rgba(165, 165, 165, 0.6)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 6,
                "width": 714,
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 16,
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexWrap": "nowrap",
                    "height": 96,
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  animate={null}
                  customAnimate={null}
                  onReady={false}
                  style={
                    Object {
                      "aspectRatio": 1,
                      "backgroundColor": "#A5A5A5",
                      "height": "100%",
                      "resizeMode": "cover",
                      "width": "100%",
                    }
                  }
                >
                  <Image
                    onLoad={[Function]}
                    source={
                      Array [
                        Object {
                          "cache": "force-cache",
                          "uri": "https://www.placecage.com/c/250/250",
                          "url": "https://www.placecage.com/c/250/250",
                        },
                      ]
                    }
                    style={
                      Object {
                        "aspectRatio": 1,
                        "backgroundColor": "#A5A5A5",
                        "height": "100%",
                        "opacity": 0,
                        "resizeMode": "cover",
                      }
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "paddingHorizontal": 16,
                      "paddingVertical": 16,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "stretch",
                        "flexDirection": "row",
                        "flexWrap": "nowrap",
                        "height": 21,
                        "justifyContent": "space-between",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#303030",
                          "fontFamily": "InterUI-Medium",
                          "fontSize": 14,
                          "lineHeight": 21,
                        }
                      }
                    >
                      Cincinnati Campus
                    </Text>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                          "justifyContent": "flex-end",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#505050",
                            "fontFamily": "InterUI-Bold",
                            "fontSize": 12,
                            "lineHeight": 18,
                          }
                        }
                      >
                        2038
                        mi
                      </Text>
                    </View>
                  </View>
                  <Text
                    style={
                      Object {
                        "color": "#505050",
                        "fontFamily": "InterUI-Bold",
                        "fontSize": 12,
                        "lineHeight": 18,
                      }
                    }
                  >
                    120 E 8th St
Cincinnati, OH 45202-2118
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "paddingHorizontal": 16,
          "paddingVertical": 16,
        }
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          bordered={false}
          disabled={false}
          pill={false}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#17B582",
              "borderColor": "#17B582",
              "borderRadius": 16,
              "borderWidth": 2,
              "elevation": 2,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "center",
              "opacity": 1,
              "overflow": "hidden",
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "InterUI-Bold",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
          >
            Custtom buttonTitle
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
